edx_proctoring/utils.py

```diff
@@ -231,3 +231,30 @@ def obscured_user_id(user_id, *extra):
     obs_hash.update(six.text_type(user_id))
     obs_hash.update(u''.join(six.text_type(ext) for ext in extra))
     return obs_hash.hexdigest()
+
+
+def has_due_date_passed(due_datetime):
+    """
+    return True if due date is lesser than current datetime, otherwise False
+    and if due_datetime is None then we don't have to consider the due date for return False
+    """
+
+    if due_datetime:
+        return due_datetime <= datetime.now(pytz.UTC)
+    return False
+
+
+def verify_and_add_wait_deadline(context, exam_due_date):
```

Comment:
nit: `exam_has_due_date`?

Reply:
exam_has_due_date doesn't signify the work being done in the function. This function is basically confirming if the grace period is active and the waiting banner should be shown or not.